File: dayflow-hrms/frontend/src/services/notificationService.js
import api from './api';

export const notificationService = {
  getMyNotifications: () => api.get('/notifications'),
  getUnreadNotifications: () => api.get('/notifications/unread'),
  markAsRead: (id) => api.put(`/notifications/${id}/read`),
  markAllAsRead: () => api.put('/notifications/read-all')
};

File: dayflow-hrms/backend/src/main/java/com/dayflow/hrms/service/NotificationService.java
package com.dayflow.hrms.service;

import com.dayflow.hrms.dto.notification.NotificationDto;
import com.dayflow.hrms.enums.NotificationType;
import java.util.List;

public interface NotificationService {
    NotificationDto createNotification(Long userId, String title, String message, NotificationType type);
    void notifyAllAdmins(String title, String message, NotificationType type);
    List<NotificationDto> getUserNotifications(Long userId);
    List<NotificationDto> getUnreadNotifications(Long userId);
    NotificationDto markAsRead(Long notificationId, Long userId);
    void markAllAsRead(Long userId);
    long getUnreadCount(Long userId);
}
